pkg/bundle/internal/truststore/types_test.go

```diff
@@ -146,3 +143,60 @@ func Test_certAlias(t *testing.T) {
 		t.Fatalf("expected alias to be %q but got %q", expectedAlias, alias)
 	}
 }
+
+func Test_manual_read_JKS(t *testing.T) {
+	t.Skip("these tests require installation of Java to run")
+
+	tests := map[string]struct {
+		keytool string
+	}{
+		"Java 21": {
+			keytool: "/usr/lib/jvm/java-21-openjdk-amd64/bin/keytool",
+		},
+		"Java 17": {
+			keytool: "/usr/lib/jvm/java-17-openjdk-amd64/bin/keytool",
+		},
+		"Java 11": {
+			keytool: "/usr/lib/jvm/java-11-openjdk-amd64/bin/keytool",
+		},
+		"Java 8": {
+			keytool: "/usr/lib/jvm/java-8-openjdk-amd64/bin/keytool",
+		},
+	}
+
+	bundle := dummy.JoinCerts(dummy.TestCertificate1, dummy.TestCertificate2, dummy.TestCertificate3)
+
+	certPool := util.NewCertPool()
+	if err := certPool.AddCertsFromPEM([]byte(bundle)); err != nil {
+		t.Fatalf("didn't expect an error but got: %s", err)
+	}
+
+	encoder := NewPKCS12Encoder(v1alpha1.DefaultJKSPassword, v1alpha1.LegacyRC2PKCS12Profile)
+	store, err := encoder.Encode(certPool)
+	if err != nil {
+		t.Fatalf("didn't expect an error but got: %s", err)
+	}
+
+	f, err := os.Create(path.Join(t.TempDir(), "cacerts"))
+	if err != nil {
+		t.Fatalf("didn't expect an error but got: %s", err)
+	}
+	defer f.Close()
+	_, err = f.Write(store)
+	if err != nil {
+		t.Fatalf("didn't expect an error but got: %s", err)
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			cmd := exec.Command(test.keytool, "-list", "-keystore", f.Name(), "-storepass", v1alpha1.DefaultJKSPassword) // #nosec G204
+			out, err := cmd.CombinedOutput()
+			if err != nil {
+				t.Fatalf("cmd.Run() failed with %s\n", err)
```

Comment:
nitpick: Don't need the newline in Fatalf!

```suggestion
				t.Fatalf("cmd.Run() failed with %s", err)
```